thunder/distributed/transforms/fsdp_v2.py

```diff
@@ -166,3 +170,41 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
                 provenance="fsdp pass with unpadding",
             )
         return prologue_trace, new_computation_trace, epilogue_trace
+
+    def transform_state_dict(
```

Comment:
I would like to make this per submodule (similar to `transform_state_dict_for_submodule` but the reverse for the option to deal with it incrementally.

Reply:
I'm yet to realize why "state_dict_for_submodule" is important, not vanilla state dict

Reply:
and what is the way to reconstruct `state_dict` from state_dict_per_module?

Reply:
We don't want to materialize the entire state dict for models where the state dict does not fit into memory untransformed. Pretty much the same thing as why we do transform_state_dict_for_submodule or why the classic fsdp materialization did that submodule by submodule.
We can reconstruct `state_dict` as similar to what we do in `load_original_state_dict`, but here is the thing: in litgpt we have a thing for incrementally saving state dicts, so we can adapt that to get the state dict straight to disk: https://github.com/Lightning-AI/litgpt/blob/1bd860000a8daa3bbd263ba5eed9911189db0dbd/litgpt/utils.py#L238